src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+
+    public RightTriangle(String color, double firstLeg, double secondLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    @Override
+    public String toString() {
+        return "Figure: right triangle, area: " + calculateArea() + " sq. units, "
+                + "first leg: " + firstLeg + " units, "
+                + "second leg: " + secondLeg + " units, "
+                + "color: " + getColor();
+    }
+
+    @Override
+    public double calculateArea() {
+        return 0.5 * firstLeg * secondLeg;
+    }
+
+    @Override
+    public void draw(Figure figure) {

```

Comment:
The 'draw' method should not take a 'Figure' as a parameter since it's already in the context of the 'Figure' it needs to draw. It can use 'this' to refer to the current object.